moquette-0.17/broker/src/main/java/io/moquette/broker/BrokerConfiguration.java

```diff
@@ -18,35 +18,65 @@
 import io.moquette.BrokerConstants;
 import io.moquette.broker.config.IConfig;
 
+import java.util.Locale;
+
 class BrokerConfiguration {
 
     private final boolean allowAnonymous;
     private final boolean allowZeroByteClientId;
     private final boolean reauthorizeSubscriptionsOnConnect;
-    private final boolean immediateBufferFlush;
+    private final int bufferFlushMillis;
     private final boolean peerCertificateAsUsername;
 
     BrokerConfiguration(IConfig props) {
         allowAnonymous = props.boolProp(BrokerConstants.ALLOW_ANONYMOUS_PROPERTY_NAME, true);
         allowZeroByteClientId = props.boolProp(BrokerConstants.ALLOW_ZERO_BYTE_CLIENT_ID_PROPERTY_NAME, false);
         reauthorizeSubscriptionsOnConnect = props.boolProp(BrokerConstants.REAUTHORIZE_SUBSCRIPTIONS_ON_CONNECT, false);
-        immediateBufferFlush = props.boolProp(BrokerConstants.IMMEDIATE_BUFFER_FLUSH_PROPERTY_NAME, false);
         peerCertificateAsUsername = props.boolProp(BrokerConstants.PEER_CERTIFICATE_AS_USERNAME, false);
+
+        // BUFFER_FLUSH_MS_PROPERTY_NAME has precedence over the deprecated IMMEDIATE_BUFFER_FLUSH_PROPERTY_NAME
+        final String bufferFlushMillisProp = props.getProperty(BrokerConstants.BUFFER_FLUSH_MS_PROPERTY_NAME);
+        if (bufferFlushMillisProp != null && !bufferFlushMillisProp.isEmpty()) {
+            switch (bufferFlushMillisProp.toLowerCase(Locale.ROOT)) {
+                case "immediate":
+                    bufferFlushMillis = BrokerConstants. IMMEDIATE_BUFFER_FLUSH;
+                    break;
+                case "full":
+                    bufferFlushMillis = BrokerConstants.NO_BUFFER_FLUSH;
+                    break;
+                default:
+                    final String errorMsg = String.format("Can't state value of %s property. Has to be 'immediate', " +
+                        "'full' or a number >= -1, found %s", BrokerConstants.BUFFER_FLUSH_MS_PROPERTY_NAME, bufferFlushMillisProp);
+                    try {
+                        bufferFlushMillis = Integer.parseInt(bufferFlushMillisProp);
+                        if (bufferFlushMillis < -1) {
+                            throw new IllegalArgumentException(errorMsg);
+                        }
+                    } catch (NumberFormatException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)